assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -2246,4 +2252,82 @@ public SimplePeerAssessmentItem(PeerAssessmentItem item) {
                     .collect(Collectors.toList());
         }
     }
+
+    /**
+     * Helper method to find a valid submitter ID for LTI from an AssignmentSubmission
+     */
+    private String findValidSubmitterIdForLTI(AssignmentSubmission as, Assignment assignment) {
+        Set<AssignmentSubmissionSubmitter> submitters = as.getSubmitters();
+        
+        if (!submitters.isEmpty()) {
+            for (AssignmentSubmissionSubmitter submitter : submitters) {
+                if (submitter.getSubmittee() != null && submitter.getSubmittee()) {
+                    String submitterId = submitter.getSubmitter();
+                    log.info("Found actual submitter (submittee=true): {}", submitterId);
+                    return submitterId;
+                }
+            }
+        }
+        
+        if (assignment.getIsGroup() && StringUtils.isNotBlank(as.getGroupId())) {
+            try {
+                Site site = siteService.getSite(assignment.getContext());
+                Group group = site.getGroup(as.getGroupId());
+                if (group != null && !group.getUsers().isEmpty()) {
+                    String submitterId = group.getUsers().iterator().next();
+                    log.info("Using first user from group: {}", submitterId);
+                    return submitterId;
+                }
+            } catch (Exception e) {
+                log.warn("Error getting users from group: {}", e.getMessage());
+            }
+        }
+        
+        if (!submitters.isEmpty()) {
+            String submitterId = submitters.iterator().next().getSubmitter();
+            log.info("Using first submitter from list: {}", submitterId);
+            return submitterId;
+        }
+        
+        log.warn("Could not find a valid submitter ID for assignment: {}", assignment.getId());
+        return null;
+    }
+    
+    /**
+     * Helper method to find a valid submitter ID for LTI from a submission Map
+     */
+    private String findValidSubmitterIdFromMap(Map<String, Object> submission, Site site) {
```

Comment:
the seems identical to above? if your looking for who submitted the submissions use 
```
assignmentService.getSubmissionSubmittee(as);
```